### PR TITLE
Added HFI "load" route

### PR DESCRIPTION
**Pull request by Sybrand**
This is the 1st step in getting rid of the HFIResulRequest object and /api/hfi-calc route.

- [x] Add and wire up new route: /api/hfi-calc/load
- [x] Write back-end tests for load
- [x] Write front-end tests for load
- [x] `poetry update` (because why not!)

# Test Links:
[Percentile Calculator](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1784.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/routers/hfi_calc.py [hunk 1]
```diff
@@ -62,6 +63,72 @@ def extract_selected_stations(request: HFIResultRequest) -> List[int]:
     return stations_codes
 
 
+@router.post("/load", response_model=HFIResultResponse)
+async def load_hfi_result(request: HFILoadResultRequest,
+                          response: Response,
+                          token=Depends(authentication_required)):
+    """ Given a fire centre id (and optionally a start date), load the most recent HFIResultRequst.
```
Comment (conbrad) [inline api/app/routers/hfi_calc.py:70]: HFIResultRequst -> HFIResultRequest

**Review by conbrad — APPROVED** [review]
LGTM I just forget the overarching idea of what the full set of changes are. Is this endpoint just for the initial page load? Then there will be individual routes for the different kinds of updates?

File: api/app/schemas/hfi_calc.py [hunk 2]
```diff
@@ -144,6 +144,12 @@ class StationInfo(BaseModel):
     fuel_type_id: int
 
 
+class HFILoadResultRequest(BaseModel):
+    """ Request to load the HFI Calculator. """
+    start_date: Optional[date]
```
Comment (andrea-williams) [inline api/app/schemas/hfi_calc.py:149]: Just want to confirm for my own understanding - if a fire centre has a prep sheet stored in our DB where the prep sheet's `start_date` matches this `start_date`, the result is returned? Or is it that any date in the prep sheet's prep period that includes this `start_date` returns a result?

**Review by andrea-williams — APPROVED** [review]

File: api/app/routers/hfi_calc.py (hunk 1 above)
Comment (andrea-williams) [reply api/app/routers/hfi_calc.py:70]: sometimes @conbrad eases my workload of pointing out all of Sy's spellign misteaks. I like it!

File: api/app/schemas/hfi_calc.py (hunk 2 above)
Comment (Sybrand) [reply api/app/schemas/hfi_calc.py:149]: Not period - has to match the start date.

**Comment by Sybrand** [comment]
> LGTM I just forget the overarching idea of what the full set of changes will be. Is this endpoint just for the initial page load? Then there will be individual routes for the different kinds of updates?

Each update will have it's own individual route. I have as tasks:
 Extract "load" action.
 Extract "change fire starts" action.
 Extract "select/deselect station" action.
 Extract "change prep period" action.
 Extract "change fuel type" action.
 Remove save changes button.

So for example, for "change fire starts"  it's going to be: "for this fire centre, for this prep start date, for this planning area, for this day, set the fire starts to x".

File: api/app/routers/hfi_calc.py (hunk 1 above)
Comment (Sybrand) [reply api/app/routers/hfi_calc.py:70]: I cnt evn c thu prblm.

**Pull request merged**